### PR TITLE
add Pony

### PR DESCRIPTION
**Pull request by ubnt-intrepid**
Pony is an object-oriented, actor-model language focus on capabilities-secure programming (details are [here](https://github.com/ponylang/ponyc)).

### CODE REVIEW

File: test/run.py [hunk 1]
```diff
@@ -255,6 +261,7 @@ def register():
     test_generic(name='ocaml', test_file='test.ml', expected='Hello, world!\n', with_head=True)
     test_generic(name='go', test_file='test.go', expected='hello\n', with_head=True)
     test_generic(name='sbcl', test_file='test.lisp', expected='hello', with_head=True)
+    test_pony_head()
```
Comment (melpon) [inline test/run.py]: Add a test for `pony-0.11.3`.
Maybe, you can use `test_generic` function: `test_generic(name='pony', test_file='test/main.pony', excepted='hello\n', with_head=True)`.

File: cattleshed-conf/compilers.py [hunk 2]
```diff
@@ -1789,6 +1789,37 @@ def make_bash(self):
...
+                display_name = 'pony'
+                version_command = ['/bin/echo', '{cv}']
+
+            compilers.append(format_value({
+                'name': 'pony-{cv}',
+                'displayable': True,
+                'language': 'Pony',
+                'output-file': 'prog.pony',
+                'compiler-option-raw': True,
+                'compile-command': ['/bin/bash', '-c', "mkdir -p prog && cp ./prog.pony ./prog/main.pony && /opt/wandbox/pony-{cv}/bin/ponyc ./prog"],
```
Comment (melpon) [inline cattleshed-conf/compilers.py]: Nice hack! 👍

File: build/pony-head/install.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+. ../init.sh
+
+PREFIX=/opt/wandbox/pony-head
+
+# get LLVM
```
Comment (melpon) [inline build/pony-head/install.sh]: Change directory to `~/`

File: test/run.py (hunk 1 above)
Comment (melpon) [reply test/run.py]: And integration tests failed.
```
$ ./run-test.sh 'pony-*'
2cd324c9b573f6328634de8cd78b8f46864da34145647d25e25eb9a657a6a935
wait: kennel port...
opened
pony-0.11.3: {u'status': u'1', u'program_error': u'execve: Permission denied\n', u'program_message': u'execve: Permission denied\n', u'compiler_error': u'Building builtin -> /opt/wandbox/pony-0.11.3/lib/pony/0.11.3/packages/builtin\nBuilding ./prog -> /home/jail/prog\nGenerating\n Reachability\n Selector painting\n Data prototypes\n Data types\n Function prototypes\n Functions\n Descriptors\nOptimising\nWriting ./prog.o\nLinking ./prog1\nWarning: environment variable $CC undefined, using cc as the linker\n', u'compiler_message': u'Building builtin -> /opt/wandbox/pony-0.11.3/lib/pony/0.11.3/packages/builtin\nBuilding ./prog -> /home/jail/prog\nGenerating\n Reachability\n Selector painting\n Data prototypes\n Data types\n Function prototypes\n Functions\nDescriptors\nOptimising\nWriting ./prog.o\nLinking ./prog1\nWarning: environment variable $CC undefined, using cc as the linker\n'}
pony-0.11.3: failed ()
pony-head: {u'status': u'1', u'program_error': u'execve: Permission denied\n', u'program_message': u'execve: Permission denied\n', u'compiler_error': u'Building builtin -> /opt/wandbox/pony-head/lib/pony/0.11.3-f5dc43d/packages/builtin\nBuilding ./prog -> /home/jail/prog\nGenerating\n Reachability\n Selector painting\n Data prototypes\n Data types\n Function prototypes\n Functions\n Descriptors\nOptimising\nWriting ./prog.o\nLinking ./prog1\nWarning: environment variable $CC undefined, using cc as the linker\n', u'compiler_message': u'Building builtin -> /opt/wandbox/pony-head/lib/pony/0.11.3-f5dc43d/packages/builtin\nBuilding ./prog -> /home/jail/prog\nGenerating\n Reachability\n Selector painting\n Data prototypes\n Data types\n Function prototypes\nFunctions\nDescriptors\nOptimising\nWriting ./prog.o\nLinking ./prog1\nWarning: environment variable $CC undefined, using cc as the linker\n'}
pony-head: failed ()
test-server
66a5e636a918
2cd324c9b573
```

Generally, "Permission denied" is occurred when `compile-command` or `run-command` don't have an execute permission.

**Comment by ubnt-intrepid** [comment]
I fixed some issues in `build/pony-*/install.sh` pointed out by review.

Removing problems which cause test failures has not completed yet. Please wait a moment...

File: cattleshed-conf/compilers.py [hunk 4]
```diff
@@ -1806,7 +1806,7 @@ def make_pony(self):
                 'language': 'Pony',
                 'output-file': 'prog.pony',
                 'compiler-option-raw': True,
-                'compile-command': ['/bin/bash', '-c', "mkdir -p prog && cp ./prog.pony ./prog/main.pony && /opt/wandbox/pony-{cv}/bin/ponyc ./prog"],
+                'compile-command': ['/bin/bash', '-c', "mkdir -p /tmp/prog && cp ./prog.pony /tmp/prog/main.pony && /opt/wandbox/pony-{cv}/bin/ponyc /tmp/prog"],
```
Comment (ubnt-intrepid) [inline cattleshed-conf/compilers.py]: `ponyc` outputs compiled binary as name `./prog1` (`./prog` is already existed), that is the reason why integration test fails with error `execve: Permission denied`.

**Comment by ubnt-intrepid** [comment]
OK, I have finished to elimininate issues which cause integration test failures!

**Comment by melpon** [comment]
Thank you for your contribution! I will ready pony to Wandbox soon.

**Comment by melpon** [comment]
Done: http://melpon.org/wandbox/permlink/9935fcpN94OxemdM

**Pull request merged**